### DISCUSSION TITLE
Does COMPlus_Thread_UseAllCpuGroups has an effect on NUMA cores

### DISCUSSION THREAD

**Discussion by msedi** [dotnet/runtime#58116] [Q&A]
We have an AMD Threadripper with 2 NUMA cores (in total: 128 cores) and try to run a simple program (see below), but we can only see that one NUMA core is used.  Environment.ProcessorCount only shows 64 which is somehow OK and has already been discussed, but the question is does `COMPlus_Thread_UseAllCpuGroups `have an effect at all or is there currently no way to use all NUMA cores?

 ```cs
            ParallelOptions options = new ParallelOptions()
            {
                MaxDegreeOfParallelism = Environment.ProcessorCount
            };


            Parallel.For(0, 100, options, idx =>
            {
                Random r = new Random();
                double result = 0;

                for (long i = 0; i < long.MaxValue; i++)
                {
                    unchecked { result += Math.Log(r.NextDouble()); };
                }

                Console.WriteLine($"Result {result}");
            });
```

**Comment by msedi** [comment]
Just wanted to follow up on this if there is an running example of using NUMA processing. Does Parallel.For support it? I'm just checking if my machine has a bad setting/condition or if I'm doing something wrong. There are little/no docs around this. Other than setting the COMPlus_Thread_UseAllCpuGroups  or DOTNET_Thread_UseAllCpuGroups I cannot see some usage in some discussion.

**Comment by danmoseley** [comment]
@janvorli  maybe can answer this one
